test/unit/data-types.test.ts

```diff
@@ -0,0 +1,81 @@
+import { DataTypes } from '@sequelize/core';
+import type { StringifyOptions } from '@sequelize/core/_non-semver-use-at-your-own-risk_/dialects/abstract/data-types.js';
+import { expect } from 'chai';
+import identity from 'lodash/identity';
+import { sequelize } from '../support';
+
+const dialect = sequelize.dialect;
+
+const stringifyOptions: StringifyOptions = {
+  dialect,
+  escape: identity,
+};
+
+describe('DataTypes', () => {
+  const now = new Date();
+  const nowString = now.toISOString();
+  const dateType = DataTypes.DATE().toDialectDataType(dialect);
+  const dateOnlyType = DataTypes.DATEONLY().toDialectDataType(dialect);
+
+  describe('DATE/DATEONLY Validate and Stringify', () => {
+    it('DATE should validate a Date as normal', () => {
+      expect(() => dateType.validate(now)).not.to.throw();
+      expect(() => dateType.validate(nowString)).not.to.throw();
+    });
+
+    if (dialect.name === 'postgres') {
```

Comment:
Can we make this into a `dialect.supports` field as well?

Reply:
it even already exists, done